commerce/domain/shipping/splitshipment/SplitShipmentUnitTest.cls

```diff
@@ -0,0 +1,303 @@
+                .withDeliverToState('MA')
+                .withDeliverToStreet('2 Patriot Place')
+                .withDeliverToCompanyName('New England Patriots')
+                .withDeliveryAddressId('8lW456789012679AAA')
+                .build();
+
+		List<CartExtension.ItemArrange> itemArrangeList = new List<CartExtension.ItemArrange>();
+        itemArrangeList.add(itemArrange);
+
+        CartExtension.ItemArrangementRequest request = new CartExtension.ItemArrangementRequest.Builder()
+                .withCart(cart)
+                .withItemArrangeList(itemArrangeList)
+                .build();
+
+        // Act
+        splitShipmentCallsSuper.arrangeItems(request);
+
+        // Assert
+        Assert.isTrue(cart.getName().contains('Default Split Shipment Service was invoked'));
+    }
+
+    @IsTest
+    public static void splitShipmentApexWithDeliveryGroupIdAndProductId() {
+        // Arrange
+        CartExtension.Cart cart = CartExtension.CartTestUtil.createCart();
+
+        CartExtension.ItemArrange itemArrange = new CartExtension.ItemArrange.Builder()
+                .withProductId('01txx0000006iP2AAI')
+                .withQuantity(Decimal.valueOf(5))
+                .withDeliveryGroupId(cart.getCartItems().get(0).getCartDeliveryGroup().getId())
+                .build();
+
+		List<CartExtension.ItemArrange> itemArrangeList = new List<CartExtension.ItemArrange>();
+        itemArrangeList.add(itemArrange);
+
+        CartExtension.ItemArrangementRequest request = new CartExtension.ItemArrangementRequest.Builder()
+                .withCart(cart)
+                .withItemArrangeList(itemArrangeList)
+                .build();
+
+        // Act
+        splitShipmentSample.arrangeItems(request);
+
+        // Assert
+        Assert.isTrue(cart.getCartDeliveryGroups().size() == 1);
+        Assert.isTrue(cart.getCartItems().size() == 2);
+        Assert.isTrue(cart.getCartItems().get(1).getQuantity() == 5.0);
+    }
+
+    @IsTest
+    public static void splitShipmentApexWithDeliveryGroupIdAndCartItemId() {
+        // Arrange
+        CartExtension.Cart cart = CartExtension.CartTestUtil.createCart();
+
+        CartExtension.ItemArrange itemArrange = new CartExtension.ItemArrange.Builder()
+                .withCartItemId(cart.getCartItems().get(0).getId())
+                .withQuantity(Decimal.valueOf(5))
+                .withDeliveryGroupId(cart.getCartItems().get(0).getCartDeliveryGroup().getId())
+                .build();
+
+		List<CartExtension.ItemArrange> itemArrangeList = new List<CartExtension.ItemArrange>();
+        itemArrangeList.add(itemArrange);
+
+        CartExtension.ItemArrangementRequest request = new CartExtension.ItemArrangementRequest.Builder()
+                .withCart(cart)
+                .withItemArrangeList(itemArrangeList)
+                .build();
+
+        // Act
+        splitShipmentSample.arrangeItems(request);
+
+        // Assert
```

Comment:
apexdoc needed to explain why we are asserting what we are asserting. Without deep knowledge of ItemArrangement it is impossible to understand the expectations. But this sample is supposed to serve as documentation - so let's make such things clear.